miner/worker.go

```diff
@@ -1076,16 +1081,89 @@ func (w *worker) prepareWork(genParams *generateParams) (*environment, error) {
 // into the given sealing block. The transaction selection and ordering strategy can
 // be customized with the plugin in the future.
 func (w *worker) fillTransactions(interrupt *int32, env *environment) error {
-	// Split the pending transactions into locals and remotes
-	// Fill the block with all available pending transactions.
+	// First pull Electroneum Ltd transactions out of 'pending' if flag --prioritiseElectroneum was passed by
+	// cli argument and then Split the pending transactions into locals and remotes
+	// pending is a local copy of the pool not a pointer, so it's ok to strip out Electroneum tx from localTxs and
+	// remoteTxs before processing them separately and avoid double counting.
 	pending := w.eth.TxPool().Pending(true)
+
+	electroneumTxs := make(map[common.Address]types.Transactions)
+	if w.config.PrioritiseElectroneum {
+
+		//todo: move this into worker or somewhere to save doing this every time we fill a block
+		//testing keys only
+		// private key : 8dd88ba44afa275e0dd9be1d633732242d84ea955e453b28719d17897035b4c2
+		//NB the 0x04 is to tell us that the key is uncompressed 64bytes +1 byte
+		etnSigningPubKey := "04" + "d6e7d6da18e8cfb9cf20bbb7182ac36deb9b77f2c617f3698fc12aebf22ee9b00060775349a2312813818a9c7a0a4b143fddfa36611129daf01c26bc9080d418"
```

Comment:
perhaps for a future release, but would be good to get the public keys from an endpoint if we decide to expand this feature to key partners. code needs to be adjusted to handle a list of public keys

Reply:
Will be for a future release yeah - I thought about implementing what you have said from the get-go but just went for one variable param for expedience just for now. There might also be a priority hierarchy requiring nested arrays. 
Will move this key into the config also if there is time before release